src/Nethermind/Nethermind.Specs/ChainSpecStyle/ChainSpecBasedSpecProvider.cs

```diff
@@ -170,28 +171,12 @@ private void BuildTransitions()
 
         public IReleaseSpec GenesisSpec => _transitions.Length == 0 ? null : _transitions[0].Release;
 
-        public IReleaseSpec GetSpec(long blockNumber)
-        {
-            if (_transitions.Length == 0)
-            {
-                return null;
-            }
-
-            IReleaseSpec spec = _transitions[0].Release;
-            for (int i = 1; i < _transitions.Length; i++)
-            {
-                if (blockNumber >= _transitions[i].BlockNumber)
-                {
-                    spec = _transitions[i].Release;
-                }
-                else
-                {
-                    break;
-                }
-            }
-
-            return spec;
-        }
+        public IReleaseSpec GetSpec(long blockNumber) =>
+            _transitions.TryGetSearchedItem(blockNumber,
```

Comment:
unnecessary lambda allocation on GetSpec which is called very often

Reply:
fixed by static method